community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexProviderMap.java

```diff
@@ -20,18 +20,44 @@
 package org.neo4j.kernel.impl.api.index;
 
 import java.util.function.Consumer;
-import java.util.function.Function;
 
 import org.neo4j.kernel.api.index.IndexProvider;
 
-public interface IndexProviderMap extends Function<IndexProvider.Descriptor,IndexProvider>
+/**
+ * Contains mapping from {@link IndexProvider.Descriptor} or provider name to {@link IndexProvider}.
+ */
+public interface IndexProviderMap
 {
-    @Override
-    IndexProvider apply( IndexProvider.Descriptor providerDescriptor ) throws IndexProviderNotFoundException;
+    /**
+     * Looks up and returns the {@link IndexProvider} for the given {@link IndexProvider.Descriptor}.
+     *
+     * @param providerDescriptor the descriptor identifying the {@link IndexProvider}.
+     * @return the {@link IndexProvider} with the given {@link IndexProvider.Descriptor}.
+     * @throws IndexProviderNotFoundException if no such {@link IndexProvider} was found.
+     */
+    IndexProvider lookup( IndexProvider.Descriptor providerDescriptor ) throws IndexProviderNotFoundException;
```

Comment:
`lookup` way better than `apply`! Gonna suggest `get` as well because that is what a map usually provides. Take it or leave it ☺️